### PR TITLE
[FedLLM] support client/aggregator test option; improve zero3 model loading; code cleanup

### PR DESCRIPTION
**Pull request by fedml-zijianhu**

### CODE REVIEW

**Review by chaoyanghe — COMMENTED** [review]
See my review. Once you finished the fix, I will merge.

File: python/app/fedllm/main_fedllm.py [hunk 1]
```diff
@@ -37,8 +40,9 @@ def _parse_args(args: Arguments) -> Arguments:
     if args.role == "client":
         if hasattr(args, "client_dataset_path"):
             args.dataset_path = args.client_dataset_path
-        # disable logging for client
-        setattr(args, "report_to", "none")
+        if not getattr(args, "is_client_test", False):
+            # disable logging for client when not testing on client
```
[the remark this answers is not in the document]
Comment (fedml-zijianhu) [reply python/app/fedllm/main_fedllm.py:47]: This is a huggingface configuration. It controls how huggingface `Trainer` save the log. I will add a comment to explain this

File: python/app/fedllm/main_fedllm.py [hunk 2]
```diff
@@ -140,18 +145,20 @@ def set_model_params(self, model_parameters) -> None:
...
 
         self.trainer.train_dataset = train_data
 
-        if getattr(self.args, "round_idx", -1) > 0:
+        if self.round_idx > 0:
             # TODO: verify model, model_wrapped, deepspeed, optimizer, lr_scheduler after reset
             # turn off TrainingArguments.deepspeed to avoid duplicated initializations
             self.trainer.args.deepspeed = None
 
+            # TODO: remove once FedML integrated the change
```
[the remark this answers is not in the document]
Comment (fedml-zijianhu) [reply python/app/fedllm/main_fedllm.py:186]: Done. You can see it here https://github.com/FedML-AI/FedML/issues/1096

File: python/app/fedllm/fedml_config/fedml_config.yaml [hunk 3]
```diff
@@ -56,6 +56,8 @@ train_args:
 
 validation_args:
   frequency_of_the_test: 1
+  is_aggregator_test: true
+  is_client_test: false
```
[the remark this answers is not in the document]
Comment (fedml-zijianhu) [reply python/app/fedllm/fedml_config/fedml_config.yaml]: fixed in https://github.com/FedML-AI/FedML/blob/c485e575967334d8cc17ea13cb9826d770796272/python/app/fedllm/fedml_config/fedml_config.yaml#L59

File: python/app/fedllm/main_fedllm.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (fedml-zijianhu) [reply python/app/fedllm/main_fedllm.py]: fixed in https://github.com/FedML-AI/FedML/blob/c485e575967334d8cc17ea13cb9826d770796272/python/app/fedllm/main_fedllm.py#L46-L48

File: python/app/fedllm/main_fedllm.py [hunk 4]
```diff
@@ -275,7 +316,7 @@ def main(args: Arguments) -> None:
     dataset = transform_data_to_fedml_format(args, dataset)
 
     # FedML trainer
-    trainer = LLMTrainer(model=model, args=args, tokenizer=tokenizer, model_args=model_args)
+    trainer = LLMTrainer(model=model, args=args, tokenizer=tokenizer, model_args=model_args, test_dataset=test_dataset)
```
[the remark this answers is not in the document]
Comment (fedml-zijianhu) [reply python/app/fedllm/main_fedllm.py:387]: fixed here https://github.com/FedML-AI/FedML/blob/1f6e56eb6f601357cafe5aa93ae5e7aa26ec18b6/python/app/fedllm/main_fedllm.py#L326-L353

and here
https://github.com/FedML-AI/FedML/blob/1f6e56eb6f601357cafe5aa93ae5e7aa26ec18b6/python/app/fedllm/main_fedllm.py#L383-L388

[the remark this answers is not in the document]
Comment (fedml-zijianhu) [reply python/app/fedllm/main_fedllm.py:387]: This line
https://github.com/FedML-AI/FedML/blob/1f6e56eb6f601357cafe5aa93ae5e7aa26ec18b6/python/app/fedllm/main_fedllm.py#L387
will be changed to 
```python
trainer = LLMTrainer(model=model, args=args, tokenizer=tokenizer, model_args=model_args)
```
once issue https://github.com/FedML-AI/FedML/issues/1096 is fixed

**Pull request merged**